### ISSUE TITLE
setuid gets called before setgroup

### ISSUE THREAD

**Issue by daveol** [OpenLightingProject/ola#985]
This is an issue in the fedora package review, the problem seems to be in the file [common/base/Credentials.cpp](/OpenLightingProject/ola/blob/master/common/base/Credentials.cpp) and probably can get resolved by switching the methods around.... although I can't see where these methods get used other then the test case....

**Comment by peternewman** [comment]
It looks like they used to be used in the slp_server:
https://github.com/OpenLightingProject/ola/blob/master/debian/ola.bash-completion#L665

Do you want to try switching them around @daveol , possibly in the header too, and see if that does the trick for Fedora?

**Comment by daveol** [comment]
hmm... it seems like switching them around doesn't help... I have tried them around in the header file, the source file and the test file. it doesn't seem to work

**Comment by peternewman** [comment]
Can you give a bit more detail on what exactly is complaining and what error it's throwing, is there an online log file showing the error?

I take it removing setgroup fixes the error?

**Comment by daveol** [comment]
The error comes from rpmlint and is in verbose mode
```
ola.x86_64: E: missing-call-to-setgroups-before-setuid /usr/lib64/libolacommon.so.0.0.0
This executable is calling setuid and setgid without setgroups or initgroups.
There is a high probability this means it didn't relinquish all groups, and
this would be a potential security issue to be fixed. Seek POS36-C on the web
for details about the problem.
```

This is a problem because fedora requires you to fix all the rpmlint **errors**,  the warnings would be nice but are not required.

**Comment by nomis52** [comment]
For a start, does rpmlint do static or runtime checking? 

Try adding a call to setgroups(0, NULL) before setuid.

**Comment by daveol** [comment]
> For a start, does rpmlint do static or runtime checking?

[looks like static](https://github.com/rpm-software-management/rpmlint/blob/master/BinariesCheck.py#L53)

> Try adding a call to setgroups(0, NULL) before setuid.

wait... where? [here](https://github.com/OpenLightingProject/ola/blob/master/common/base/Credentials.cpp#L126)? and do you mean
```cpp
setgid(0,NULL);
```
(native)
or
```cpp
SetGID(0,NULL);
```
(ola function)

**Comment by daveol** [comment]
correction: its 
```cpp
setgroups()
```
they are complaining about. why don't we make a function for both ``` setgroups() ``` and ``` getgroups() ```

**Comment by peternewman** [comment]
Sounds like that might do the trick @daveol .

**Comment by daveol** [comment]
fixed in #987
